backend/windmill-worker/src/worker.rs

```diff
@@ -1539,14 +1539,25 @@ pub async fn run_worker<R: rsmq_async::RsmqConnection + Send + Sync + Clone + 's
             tracing::debug!("set worker busy to 1");
         }
 
+
         match next_job {
             Ok(Some(job)) => {
+
+
                 last_executed_job = None;
                 jobs_executed += 1;
 
                 tracing::debug!("started handling of job {}", job.id);
 
+
                 if matches!(job.job_kind, JobKind::Script | JobKind::Preview) {
+                    // TODO: Possible collisions?
+                    // Register here even tho not all scripts have progress.  
+                    // For optimization reasons, otherwise we should have check if progress exists every time script calls `setProgress`
+                    register_metric_for_job(
```

Comment:
it's better to check everytime we setProgress than to add it to every job, also you can do an optimization where you assume it exists and register if it doesn't